### ISSUE TITLE
Entity json-array casting not functioning as expected

### ISSUE THREAD

**Issue by chistel** [codeigniter4/CodeIgniter4#1359]
the json-array casting decodes the json data when submitting to db.
an instance is this,` [{"action_desription":"Order Approved by - \t@Staff[1]","action_time":"07-10-2018 14:14:28","action_ip":"127.0.0.1"}]` which is what is supposed to return to the db, instead it attempts submitting the raw array 
```
    ["action_desription"]=>"Order Approved by - @Staff[1]"
    ["action_time"]=> "07-10-2018 14:14:28"
    ["action_ip"]=> "127.0.0.1"
```

its issues revolves around the entity method toArray, which returns the raw data using the __get method.

**Comment by nowackipawel** [comment]
Could you please check if you've got the same problem using 'array' as a casing method? (value should be serialized then).

Could you provide code where you're trying full entity and save data?

**Comment by jim-parry** [comment]
Closed by #1373?
